mangaki/mangaki/admin.py

```diff
@@ -300,24 +300,45 @@ def make_sfw(self, request, queryset):
     def refresh_work_from_anidb(self, request, queryset):
         works = queryset.all()
 
+        # Check for works with missing AniDB AID
+        offending_works = []
         if not all(work.anidb_aid for work in works):
             offending_works = [work for work in works if not work.anidb_aid]
             self.message_user(request,
-                              "Certains de vos choix ne possèdent pas d'identifiant AniDB. "
+            "Certains de vos choix ne possèdent pas d'identifiant AniDB. "
+            "Leur rafraichissement a été omis. (Détails: {})"
+            .format(", ".join(map(lambda w: w.title, offending_works))),
+            level=messages.WARNING)
+
+        # Check for works that have a duplicate AniDB AID
+        anidb_aids = set(works.values_list('anidb_aid', flat=True))
```

Comment:
A slightly simpler manner to do this is to:

1. create a map between anidb_aid and a list of works:
  `anidb_aid_works = defaultdict(list)`
  for each work in works, `anidb_aid_works.append(myself)`
2. for each list in `anidb_aid_works` which have more than one element,
    add it to the conflicting list.

It removes all the need to do any supplementary queries, because `works` has everything you need.
It is also nothing more than `O(len(works))` in speed, if I'm not wrong.